### PR TITLE
SBS-3213 V1 geometry - alternative

### PR DESCRIPTION
**Pull request by BenChand**
Should be compared with @audunska pr when reviewing. The main difference is the use of a enum for the volume filters.

### CODE REVIEW

File: cognite/seismic/protos/v1/seismic_service_messages.proto [hunk 1]
```diff
@@ -169,10 +169,19 @@ message VolumeRequest {
         Identifier seismic = 1;
         int64 seismic_store_id = 2;  // Need to be a data manager or tenant user, not 3rd party, to access by tracestore
     }
-    LineBasedVolume volume = 3;
+    oneof filter {
+        LineBasedVolume volume = 3;
+        GeometryBasedVolume geometry = 5;
+    }
```
Comment (audunska) [inline cognite/seismic/protos/v1/seismic_service_messages.proto:175]: I didn't realize you could just put an existing field behind a oneof and be version-safe, but I guess that's how protobufs work!

File: cognite/seismic/protos/v1/seismic_service_messages.proto [hunk 2]
```diff
@@ -169,10 +169,19 @@ message VolumeRequest {
...
     bool include_trace_header = 4;
 }
 
+message GeometryBasedVolume {
+    Geometry geometry = 1;
+    InterpolationMethod interpolation_method = 2; // Required if the geometry describes a line, otherwise ignored
+    LineDescriptor z_range = 3;
+}
```
Comment (audunska) [inline cognite/seismic/protos/v1/seismic_service_messages.proto:183]: I like this much better than my approach!

File: cognite/seismic/protos/v1/seismic_service_messages.proto (hunk 1 above)
Comment (amorken) [reply cognite/seismic/protos/v1/seismic_service_messages.proto:183]: Protobufs are at heart just a concatenation of pairs of varint encoded tag numbers and values in a type-specific encoding. There's not even a defined ordering of fields in a message, this is explicitly left unspecified. This makes memory-efficient streaming serialization easier, but makes it impossible to reliably inspect the first N bytes of a large message to get a specific field, ie. for authenticating messages in a proxy or whatever. =)

File: cognite/seismic/protos/v1/seismic_service_messages.proto (hunk 2 above)
Comment (amorken) [reply cognite/seismic/protos/v1/seismic_service_messages.proto:191]: I like this interface too. =)

**Review by amorken — APPROVED** [review]

**Pull request merged**